### ISSUE TITLE
Fix for issue #20

### ISSUE THREAD

**Issue by mjnaylor8** [simon-the-shark/django-mapbox-location-field#22]
Hi I left a comment in issue #20 but in case you don't see that ....
-------
Issue can be fixed my moving one line of code in the init of the SpatialLocationField in forms.py. I suspect this might also be the case if you are using Location Field.

So move the line
`self.widget = MapInput(map_attrs=map_attrs, )`

below the
`super().init(*args, **kwargs)`

so the def init is now ....

```
def __init__(self, *args, **kwargs):
    map_attrs = kwargs.pop("map_attrs", None)

    super().__init__(*args, **kwargs)
    self.widget = MapInput(map_attrs=map_attrs, )
    self.error_messages = {"required": "Please pick a location, it's required", }
```

**Comment by simon-the-shark** [comment]
Hi! That's a solution to this problem. Nice job! I will include your fix in next release as soon as possible. You can also make a pull request next time.

**Comment by mjnaylor8** [comment]
I was going to make a pull request however I have also made some other changes to
1) allow pulling back all the other address fields (also hidden) e.g. country, region, location, address line 1
2) allow selection of map type
3) allow use of cursor to move map

(items 2 and 3 as per mapbox documentation)

As I have bundled all of these (bad change control I know - a github novice) it was a bit tricky to pull out one fix!

If you are interested see fork here https://github.com/mjnaylor8/django-mapbox-location-field <https://github.com/mjnaylor8/django-mapbox-location-field>

I also wanted to update to documentation but as it was a fix I wanted to get to you asap.

Great plugin by the way!

Thanks
M
